File: Algorithm/BOJ/Python/Dynamic Programming/9461.py
def solution(n):
    
    answer = [1, 1, 1, 2, 2]
    
    if n <= len(answer) - 1:
        return answer[n - 1]
    else:
        for i in range(len(answer), n):
            answer.append(answer[i - 1] + answer[i - 5])
    
    return print(answer[-1])


n = int(input())
for _ in range(n):    
    solution(int(input()))

File: Algorithm/BOJ/C++/Dynamic Programming/9461.cpp
#include <iostream>
#include <vector>

using namespace std;


long long int solution(int n) {
	
	vector<long long int> answer = { 1, 1, 1, 2, 2 };

	if (n <= answer.size()) {
		return answer[n - 1];
	} else {
		for (int i = answer.size(); i < n; i++) {
			answer.push_back(answer[i - 1] + answer[i - 5]);
		}
	}

	return answer.back();
}


int main() {
	int n, input;

	cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> input;
		cout << solution(input) << endl;		
	}

	return 0;
}
